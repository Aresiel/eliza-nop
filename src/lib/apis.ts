function rapidApiWordsApi(word: string, rapidApiKey: string) {
  return {
    method: `GET`,
    url: `https://wordsapiv1.p.rapidapi.com/words/${encodeURI(
      word
    )}/definitions`,
    headers: {
      "x-rapidapi-key": rapidApiKey,
      "x-rapidapi-host": `wordsapiv1.p.rapidapi.com`,
    },
  };
}

export async function defineWord(word: string) {
  if (!process.env.RAPID_API_KEY) {
    return null;
  }

  const { url, headers } = rapidApiWordsApi(
    word,
    process.env.RAPID_API_KEY as string
  );

  const res = await fetch(url, {
    headers,
  });

  const obj = await res.json();

  if (!obj.definitions) {
    return null;
  }

  return {
    definition: obj.definitions[0].definition,
  };
}
