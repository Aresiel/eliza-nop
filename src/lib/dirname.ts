import path from "path";
import { fileURLToPath } from "url";

export default function dirname(meta: ImportMeta) {
  return fileURLToPath(path.dirname(meta.url));
}
