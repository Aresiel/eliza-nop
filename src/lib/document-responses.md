good bye
see you
cya
> Bye bye|cya

---

uwu
> owo|uwu