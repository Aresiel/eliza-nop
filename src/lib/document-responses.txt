good bye
see you
cya
> Bye bye|cya

---

uwu
> owo|uwu

---

your mom
ur mum
you're mother
your mother
tu madre
> if you want my comeback (you can|just) (scrape|take) it off your mother's teeth
> you mean nope
> you need therapy(, clearly)?|do you want therapy\?
