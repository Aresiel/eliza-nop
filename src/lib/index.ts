export * from "./legacy-google-translate";
