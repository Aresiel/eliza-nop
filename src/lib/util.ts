export function sortByKey<T>(arr: T[], key: (o: T) => number) {
  // create cache
  const cache: Map<T, number> = new Map();

  for (const item of arr) {
    cache.set(item, key(item));
  }

  // sort
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  return arr.sort((a, b) => cache.get(a)! - cache.get(b)!);
}

export function reverse(s: string) {
  return s.split(``).reverse().join(``);
}

export function capitalize(s: string) {
  return s[0].toUpperCase() + s.slice(1);
}
