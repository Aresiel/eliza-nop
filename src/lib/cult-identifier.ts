import { awaitTick, escapeRegExp, max } from "./util";
import { nonWords } from "./nlp";

// const alphanumericSplitterRegex = /(?<=[A-Za-z])\d+|^\d+/g;
const segmentRegex = /[A-Z]{2,}(?![a-z])|[A-Z][a-z]*/g;

/**
 * Generate all possible segments from a string
 *
 * e.g. ["Hello", "World", "Its", "Me"]
 *
 * ["HelloWorldItsMe", "HelloWorldIts", "WorldItsMe", "HelloWorld", "WorldIts", "ItsMe", "Hello", "World", "Its", "Me"]
 */
function possibleSegmentGenerator(
  parts: string[],
  minmax: [number, number] = [0, Infinity]
) {
  const segments = [];

  for (let i = 0; i < parts.length; i++) {
    for (let j = i + 1; j <= parts.length; j++) {
      if (j - i > minmax[1] || j - i < minmax[0]) {
        continue;
      }
      segments.push(parts.slice(i, j).join(``));
    }
  }

  return segments;
}

// function extractParts(name: string) {
//   // space has highest priority
//   return name
//     .split(/\s+|_+|-+/)
//     .flatMap(
//       // segmentation
//       (s) => s.split(alphanumericSplitterRegex)
//     )
//     .flatMap((s) => {
//       const matches = [...s.matchAll(segmentRegex)];

//       if (matches.length === 0) matches.push([s]);

//       return matches;
//     })
//     .map((s) => s[0].normalize().toLowerCase());
// }

function capitalize(s: string) {
  return s[0]!.toUpperCase() + s.slice(1).toLowerCase();
}

function identifyAdditionalSegments(name: string) {
  // additional segments
  if (name.match(/[A-Z]/) && name.match(/[a-z]/)) {
    // identify capital parts
    const capitalParts = [...name.matchAll(/\b[A-Z]{3,}\b/g)];
    if (capitalParts.length === 1) {
      // FIA, this is for you
      return [capitalParts[0]![0]!];
    }
  }
  let m;
  if ((m = name.match(/[^\w\s]{3,}/))) {
    // if using special characters, more than 1/3 should be unique
    const icon = m[0]!;
    if (
      // shouldn't be too long
      icon.length <= 6 &&
      // should be unique
      new Set(icon).size / icon.length > 1 / 3
    )
      return [icon];
  }
  return null;
}

/**
 * Adds new property to users, be careful
 */
export async function identifyCults(users: { id: string; name: string }[]) {
  // step 1. extract identifiers
  // this is done by identifying as much identifiers as possible
  // does not care about numbers

  /**
   * `<segment, count>`
   */
  const possibleSegments = new Map<string, number>();

  for (let i = 0; i < users.length; i++) {
    const user = users[i]!;
    const name = user.name;
    const parts = [...name.matchAll(segmentRegex)]
      .map((v) => v[0].normalize().toLowerCase())
      .filter((word) => word && word.length > 1 && !nonWords[word])
      .map((word) => capitalize(word));

    if (parts.length === 0) parts.push(name);

    const segments = possibleSegmentGenerator(parts, [2, 4]);

    const additionalSegments = identifyAdditionalSegments(name);

    if (additionalSegments) {
      segments.push(...additionalSegments);
    }

    const scache = Object.create(null);

    for (const segment of segments) {
      if (scache[segment]) continue;
      scache[segment] = true;
      // lowercase identifiers
      if (segment.length <= 1) continue;

      if (!possibleSegments.has(segment)) {
        possibleSegments.set(segment, 0);
      }

      // increase by one
      possibleSegments.set(segment, possibleSegments.get(segment)! + 1);
    }

    if (i % 100 !== 0) await awaitTick;
  }

  // console.log(possibleSegments);

  // step 2. get segments with at least 2 in popularity
  for (const [segment, count] of possibleSegments) {
    // need at least 2 for a cult
    // segment must be at least 3 letters long
    if (count < 2 || segment.length < 3 || segment.match(/(.)\1\1/)) {
      possibleSegments.delete(segment);
    }
  }

  // step 3. attempt to associate users with their cults
  const indoctrinatedUsers = new Set<string>();

  const cults = [];

  const normalizeCache = Object.create(null) as Record<string, string>;
  const combineCache = Object.create(null) as Record<string, string>;
  for (const user of users) {
    normalizeCache[user.name] = user.name.normalize().toLowerCase();
    combineCache[user.name] = user.name.replace(/\W/g, ``);
  }

  // console.log([...possibleSegments.keys()].filter((x) => x.length === 3));

  while (possibleSegments.size > 0) {
    const prominentCult = max(
      possibleSegments,
      ([segment, count]) =>
        (segment.match(/[a-z]/i) ? 1000 : 0) + 100 * segment.length + count
    )!;
    const cultName = prominentCult[0]!;
    possibleSegments.delete(cultName);

    // console.log(cultName);

    // Yes, i know this is redundant, but believe it or not generating it here doesn't matter that much
    const segments = [...cultName.matchAll(segmentRegex)].map(
      (v) => v[0]
      // cult names arrive normalized already
      // .normalize().toLowerCase()
    );

    const seg = segments.map((v) => v.toLowerCase()).join(` `);
    const cultRegexB = new RegExp(`\\b${escapeRegExp(seg)}\\b`);

    const normalizedCultName = cultName.normalize().toLowerCase();

    // USA
    const isLiteralLiteral = !!cultName.match(/^[A-Z]+$/);

    let devoutAssociates = 0;

    // find associates
    const associates = users.filter((u) => {
      if (indoctrinatedUsers.has(u.id)) return false;

      if (isLiteralLiteral) {
        if (u.name === cultName) {
          devoutAssociates++;
          return true;
        }

        if (
          u.name.match(/\s/) &&
          // no need to escape regex as it is literal
          u.name.match(new RegExp(`\\b${cultName}\\b`))
        ) {
          devoutAssociates++;
          return true;
        }

        return false;
      }

      if (normalizeCache[u.name]!.includes(normalizedCultName)) {
        // segmentation stragey
        if (normalizeCache[u.name]!.split(/\W/).includes(normalizedCultName)) {
          devoutAssociates++;
          return true;
        }

        // contains both capital and lowercase letters. They must have the cult name properly
        if (u.name.match(/[A-Z]/) && u.name.match(/[a-z]/)) {
          // devout believer
          if (u.name.includes(cultName)) {
            devoutAssociates++;
            return true;
          }

          const arrname = [...u.name.matchAll(segmentRegex)]
            .map((v) => v[0].normalize().toLowerCase())
            .join(` `);
          const isMatch = !!arrname.match(cultRegexB);
          if (isMatch) devoutAssociates++;
          return isMatch;
        }

        // this is a little tricky
        if (
          u.name.match(/\s/) &&
          u.name.match(new RegExp(`\\b${cultName}\\b`))
        ) {
          // just assume that it isn't one of those ihatelife32 people
          // makes my life a little easier
          return true;
        }
        return true;
      }

      return false;
      // // you have no hope, fowl believer
      // if (!segments.some((s) => normalizeCache[u.name]!.includes(s[0]))) return;

      // // lets see if we can find a match
      // const bestMatch = bestStringMatch(u.name, cultName);

      // if (bestMatch === null) return;

      // // eh, you will do
      // if (bestMatch.length - cultName.length <= 5) return true;

      // return false;
    });

    console.log(cultName, associates);

    if (associates.length < 5 || devoutAssociates * 2 < associates.length) {
      continue;
    }

    // const exactFrequent = new Map<string, number>();

    // for (const exact of exacts) {
    //   if (exact === null) continue;

    //   if (!exactFrequent.has(exact)) {
    //     exactFrequent.set(exact, 0);
    //   }

    //   exactFrequent.set(exact, exactFrequent.get(exact)! + 1);
    // }

    // const cultRepresentative = max(exactFrequent, ([_, count]) => count)!;

    // if (cultRepresentative[1]! <= 1) {
    //   // cult is rather, indecisive
    //   possibleSegments.delete(prominentCult[0]!);
    //   continue;
    // }

    // registering cult
    cults.push({
      identifier: normalizedCultName,
      name: cultName,
      associates: associates,
    });

    // removing associates from possibleSegments
    for (const associate of associates) {
      indoctrinatedUsers.add(associate.id);

      // const associateCultAssociations = userSegmentAssociations.get(
      //   associate.id
      // );
      // if (!associateCultAssociations) continue;

      // for (const ca of associateCultAssociations) {
      //   const segment = possibleSegments.get(ca);
      //   if (!segment) continue;

      //   segment.count--;
      // }
    }

    // possibleSegments.delete(prominentCult[0]!);

    // // cleaning
    // for (const [segment, segmentInfo] of possibleSegments) {
    //   if (segmentInfo.count < 3) {
    //     possibleSegments.delete(segment);
    //   }
    // }

    await awaitTick;
  }

  return cults;
}
