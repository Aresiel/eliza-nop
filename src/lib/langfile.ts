import { readFileSync } from "fs";
import { join } from "path";
import dirname from "./dirname";
import { capitalize } from "./util";

export const langFile: Record<string, string> = JSON.parse(
  readFileSync(join(dirname(import.meta), `./langcode.json`), {
    encoding: `utf8`,
  })
);

export const reverseLangFile: Record<string, string> = Object.create(null);

for (let [key, value] of Object.entries(langFile)) {
  value = value.toLowerCase();

  const parts = value.split(/\W+/g);

  // use space joined parts
  for (let i = 0; i < Math.min(parts.length, 4); i++) {
    const firstPart = parts.slice(0, i + 1).join(` `);

    reverseLangFile[firstPart] = key;

    // extra work
    const p = parts[i]!;
    const r = (() => {
      if (p.match(/i$/)) {
        return p.replace(/i$/, ``);
      }
      if (p.match(/ese$/)) {
        return p.replace(/ese$/, ``);
      }
      if (p.match(/an$/)) {
        return p.replace(/n$/, ``);
      }
      if (p.match(/ian$/)) {
        return p.replace(/ian$/, `y`);
      }
    })();

    if (r) {
      reverseLangFile[r] = key;
    }
  }
}

/**
 * helps the bot detect proper nouns
 */
export function detectLanguageReferenceAndAutocorrect(context: string) {
  return context.replace(/\b\w+\b/, (m) => {
    const lang = reverseLangFile[m.toLowerCase()];
    if (lang) {
      return capitalize(lang);
    }
    return m;
  });
}
