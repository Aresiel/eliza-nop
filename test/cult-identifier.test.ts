import { identifyCults } from "@/lib/cult-identifier";
import { readFileLines } from "@/lib/util";

it(`identifyCults`, async () => {
  const names = readFileLines(`./test/resources/nicknames.txt`);

  const { cults } = await identifyCults(
    names.map((name) => ({
      id: name,
      name,
    }))
  );

  expect(cults).to.matchSnapshot();
});
