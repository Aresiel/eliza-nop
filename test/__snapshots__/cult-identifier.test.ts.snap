// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`identifyCults 1`] = `
[
  {
    "associates": [
      {
        "id": "FunkyCat002 ᓚᘏᗢ",
        "name": "FunkyCat002 ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCat532 but not really cat",
        "name": "FunkyCat532 but not really cat",
        "numeralDiscriminator": null,
      },
      {
        "id": "boof FunkyCat069 CuteFox069",
        "name": "boof FunkyCat069 CuteFox069",
        "numeralDiscriminator": null,
      },
      {
        "id": "Sparky FunkyCat010⚡",
        "name": "Sparky FunkyCat010⚡",
        "numeralDiscriminator": null,
      },
      {
        "id": "funkycat024",
        "name": "funkycat024",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCatgod420 ᓚᘏᗢ",
        "name": "FunkyCatgod420 ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCat006 ᓚᘏᗢ",
        "name": "FunkyCat006 ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "min funkycat042 ᓚᘏᗢ",
        "name": "min funkycat042 ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Haisong - FunkyCat005 ᓚᘏᗢ",
        "name": "Haisong - FunkyCat005 ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "orchid - FunkyCat203 ᓚᘏᗢ",
        "name": "orchid - FunkyCat203 ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCat 690 #UwUMommyToo",
        "name": "FunkyCat 690 #UwUMommyToo",
        "numeralDiscriminator": null,
      },
      {
        "id": "Akitti | FunkyCat013 ᓚᘏᗢ",
        "name": "Akitti | FunkyCat013 ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "psi | FunkyCat 420",
        "name": "psi | FunkyCat 420",
        "numeralDiscriminator": null,
      },
      {
        "id": "Crispy | FunkyCat069 ◕⩊◕",
        "name": "Crispy | FunkyCat069 ◕⩊◕",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCat004 Shork",
        "name": "FunkyCat004 Shork",
        "numeralDiscriminator": null,
      },
      {
        "id": "A.L.I.C.E | FunkyCat871",
        "name": "A.L.I.C.E | FunkyCat871",
        "numeralDiscriminator": null,
      },
      {
        "id": "jupiter the funkycat 0010 ᓚᘏᗢ",
        "name": "jupiter the funkycat 0010 ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCat130.5 -Zappkin",
        "name": "FunkyCat130.5 -Zappkin",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCat007 ᓚᘏᗢ",
        "name": "FunkyCat007 ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Val | FunkyCat(s)044 | DID sys",
        "name": "Val | FunkyCat(s)044 | DID sys",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCat001 ᓚᘏᗢ",
        "name": "FunkyCat001 ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCat666 ᓚᘏᗢ",
        "name": "FunkyCat666 ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "ains | funkycat050 ᓚᘏᗢ",
        "name": "ains | funkycat050 ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCat009ᓚᘏᗢ",
        "name": "FunkyCat009ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Funkycat~to2520",
        "name": "Funkycat~to2520",
        "numeralDiscriminator": null,
      },
      {
        "id": "Orbies the FunkyCat023 ᓚᘏᗢ",
        "name": "Orbies the FunkyCat023 ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "chartbit | FunkyCat369 ᓚᘏᗢ",
        "name": "chartbit | FunkyCat369 ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
    ],
    "identifier": "funkycat",
    "name": "FunkyCat",
  },
  {
    "associates": [
      {
        "id": "Gijscoman ᓚᘏᗢ",
        "name": "Gijscoman ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "w4v3 | ᓚᘏᗢ  | 50N1C",
        "name": "w4v3 | ᓚᘏᗢ  | 50N1C",
        "numeralDiscriminator": null,
      },
      {
        "id": "silentassasin ᓚᘏᗢ",
        "name": "silentassasin ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Biscote ᓚᘏᗢ",
        "name": "Biscote ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "clattanoid ᓚᘏᗢ",
        "name": "clattanoid ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Franz  | Red Cat Sorceress ᓚᘏᗢ",
        "name": "Franz  | Red Cat Sorceress ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Jacome Calmei ᓚᘏᗢ",
        "name": "Jacome Calmei ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Ms. ChiliFlakesᓚᘏᗢ",
        "name": "Ms. ChiliFlakesᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "lin ᓚᘏᗢ",
        "name": "lin ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "✨MLEM✨ ᓚᘏᗢ",
        "name": "✨MLEM✨ ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "ANTI-ANTI COMP SCI ᓚᘏᗢ",
        "name": "ANTI-ANTI COMP SCI ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Twilight ᓚᘏᗢ",
        "name": "Twilight ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "olanui ᓚᘏᗢ",
        "name": "olanui ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "owogy ᓚᘏᗢ",
        "name": "owogy ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Maxx ᓚᘏᗢ",
        "name": "Maxx ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Estrella ᓚᘏᗢ",
        "name": "Estrella ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "♡ Zoofy CuteFox003 ᓚᘏᗢ  😊",
        "name": "♡ Zoofy CuteFox003 ᓚᘏᗢ  😊",
        "numeralDiscriminator": null,
      },
      {
        "id": "Reigen ᓚᘏᗢ",
        "name": "Reigen ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Xpio | Proud Fermatian ᓚᘏᗢ",
        "name": "Xpio | Proud Fermatian ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Priest ᓚᘏᗢ",
        "name": "Priest ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Eku ᓚᘏᗢ 🧊😳😳",
        "name": "Eku ᓚᘏᗢ 🧊😳😳",
        "numeralDiscriminator": null,
      },
      {
        "id": "! Ale ᓚᘏᗢ",
        "name": "! Ale ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "CatGregᓚᘏᗢ",
        "name": "CatGregᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Poncho | GothCat013 ᓚᘏᗢ",
        "name": "Poncho | GothCat013 ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "nat ᓚᘏᗢ",
        "name": "nat ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "FIA - Medical Car ᓚᘏᗢ",
        "name": "FIA - Medical Car ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Wisdyyᓚᘏᗢ",
        "name": "Wisdyyᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "euler's favourite nyaᓚᘏᗢ",
        "name": "euler's favourite nyaᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
      {
        "id": "Yukki ᓚᘏᗢ",
        "name": "Yukki ᓚᘏᗢ",
        "numeralDiscriminator": null,
      },
    ],
    "identifier": "ᓚᘏᗢ",
    "name": "ᓚᘏᗢ",
  },
  {
    "associates": [
      {
        "id": "Cutefox 999 >///<",
        "name": "Cutefox 999 >///<",
        "numeralDiscriminator": null,
      },
      {
        "id": "nop CuteFox II",
        "name": "nop CuteFox II",
        "numeralDiscriminator": null,
      },
      {
        "id": "D4C CuteFox<3 420",
        "name": "D4C CuteFox<3 420",
        "numeralDiscriminator": null,
      },
      {
        "id": "Lag Deus (Insaneously CuteFox32)",
        "name": "Lag Deus (Insaneously CuteFox32)",
        "numeralDiscriminator": null,
      },
      {
        "id": "Pecky <3 (CuteFox Leader 🦊)",
        "name": "Pecky <3 (CuteFox Leader 🦊)",
        "numeralDiscriminator": null,
      },
      {
        "id": "Milan | CuteFox100",
        "name": "Milan | CuteFox100",
        "numeralDiscriminator": null,
      },
      {
        "id": "Sid | CuteFox 795 🦊",
        "name": "Sid | CuteFox 795 🦊",
        "numeralDiscriminator": null,
      },
      {
        "id": "TTT | Cutefox {010}♡",
        "name": "TTT | Cutefox {010}♡",
        "numeralDiscriminator": null,
      },
      {
        "id": "Swarnim | CuteFox007",
        "name": "Swarnim | CuteFox007",
        "numeralDiscriminator": null,
      },
    ],
    "identifier": "cutefox",
    "name": "CuteFox",
  },
  {
    "associates": [
      {
        "id": "stemer.net",
        "name": "stemer.net",
        "numeralDiscriminator": null,
      },
      {
        "id": "STEM",
        "name": "STEM",
        "numeralDiscriminator": null,
      },
      {
        "id": "chemstem249",
        "name": "chemstem249",
        "numeralDiscriminator": null,
      },
      {
        "id": "jacob_loves_stem",
        "name": "jacob_loves_stem",
        "numeralDiscriminator": null,
      },
      {
        "id": "stemmaester",
        "name": "stemmaester",
        "numeralDiscriminator": null,
      },
      {
        "id": "STEMShield",
        "name": "STEMShield",
        "numeralDiscriminator": null,
      },
      {
        "id": "highlevelsystem",
        "name": "highlevelsystem",
        "numeralDiscriminator": null,
      },
      {
        "id": "thisnewsystemisgay",
        "name": "thisnewsystemisgay",
        "numeralDiscriminator": null,
      },
      {
        "id": "STEM Power",
        "name": "STEM Power",
        "numeralDiscriminator": null,
      },
      {
        "id": "FarkasSTEM",
        "name": "FarkasSTEM",
        "numeralDiscriminator": null,
      },
      {
        "id": "STEM WEB",
        "name": "STEM WEB",
        "numeralDiscriminator": null,
      },
      {
        "id": "visit stem.help",
        "name": "visit stem.help",
        "numeralDiscriminator": null,
      },
      {
        "id": "stemaccount",
        "name": "stemaccount",
        "numeralDiscriminator": null,
      },
      {
        "id": "STEM",
        "name": "STEM",
        "numeralDiscriminator": null,
      },
      {
        "id": "thebigsadSTEM",
        "name": "thebigsadSTEM",
        "numeralDiscriminator": null,
      },
      {
        "id": "STEM?",
        "name": "STEM?",
        "numeralDiscriminator": null,
      },
      {
        "id": "Pickles | Publisher of STEM.M",
        "name": "Pickles | Publisher of STEM.M",
        "numeralDiscriminator": null,
      },
      {
        "id": "itsjustemiry",
        "name": "itsjustemiry",
        "numeralDiscriminator": null,
      },
      {
        "id": "STEM Authentication",
        "name": "STEM Authentication",
        "numeralDiscriminator": null,
      },
      {
        "id": "secret.system",
        "name": "secret.system",
        "numeralDiscriminator": null,
      },
      {
        "id": "systemsculptor",
        "name": "systemsculptor",
        "numeralDiscriminator": null,
      },
    ],
    "identifier": "stem",
    "name": "Stem",
  },
  {
    "associates": [
      {
        "id": "FunkyCow ♡MooMoo♡",
        "name": "FunkyCow ♡MooMoo♡",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCow69",
        "name": "FunkyCow69",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCow Naekk",
        "name": "FunkyCow Naekk",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCow623",
        "name": "FunkyCow623",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCow(h)",
        "name": "FunkyCow(h)",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCow 690 (M2A) #UrMomToo",
        "name": "FunkyCow 690 (M2A) #UrMomToo",
        "numeralDiscriminator": null,
      },
      {
        "id": "FunkyCow709",
        "name": "FunkyCow709",
        "numeralDiscriminator": null,
      },
    ],
    "identifier": "funkycow",
    "name": "FunkyCow",
  },
]
`;

exports[`identifyCults 2`] = `
[
  {
    "associates": [
      {
        "id": "M2A FunkyCat",
        "name": "M2A FunkyCat",
        "numeralDiscriminator": null,
      },
    ],
    "identifier": "funkycat",
    "name": "FunkyCat",
  },
  {
    "associates": [
      {
        "id": "Morbing FunkyCow",
        "name": "Morbing FunkyCow",
        "numeralDiscriminator": null,
      },
    ],
    "identifier": "funkycow",
    "name": "FunkyCow",
  },
]
`;
